tests/lib/assertions.py

```diff
@@ -62,9 +62,9 @@ def assert_error_with_optional_correlation_id(resp, code, error, correlation_id)
                 # then remove it as its a unique value that we do not know ahead of time
                 e.pop("id")
                 current_identifier_num += 1
-
             # validate the error is present
-            assert error in response_errors
+            for err in error:
+                assert str(error[err]) in str(response_errors), f"'{error[err]}' missing from response"
```

Comment:
This isn't exactly what we had before - the check before was that we expected the error to be within the list of errors - but not that every error returned by the API is the same as the expected error. Expect test failures if we relied on this logic.